python/export_util.py

```diff
@@ -235,3 +311,287 @@ def csv_query(
         data = csv_to_stream(data_list)
 
     return mgp.Record(file_path=file_path, data=data)
+
+
+def write_header(output: io.StringIO):
+    output.write('<?xml version="1.0" encoding="UTF-8"?>\n')
+    output.write(
+        '<graphml xmlns="http://graphml.graphdrawing.org/xmlns" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://graphml.graphdrawing.org/xmlns http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd">\n'  # noqa: E501
+    )
+
+
+def translate_types(variable: Any):
+    if isinstance(variable, str):
+        return "string"
+    if isinstance(variable, bool):
+        return "boolean"
+    if isinstance(variable, float):
+        return "float"
+    if isinstance(variable, int):
+        return "int"
+    raise Exception(
+        "Property values can only be primitive types or arrays of primitive types."  # noqa: E501
+    )
+
+
+def check_if_elements_same_type(variable: Any):
+    for element in variable:
+        if not isinstance(element, type(variable[0])):
+            raise Exception(
+                "If property value is a list it must consist of same typed elements."  # noqa: E501
+            )
+
+
+def get_type_string(variable: Any) -> Union[str, List[Any]]:
+    if (
+        isinstance(variable, tuple)
+    ):
+        if len(variable) == 0:
+            return ["string", "string"]
+        check_if_elements_same_type(variable)
+        return [
+            "string",
+            translate_types(variable[0]),
+        ]
+    return translate_types(variable)
+
+
+def write_keys(
+    graph: List[Union[Node, Relationship]],
+    output: io.StringIO,
+    config: mgp.Map,
+):
+    node_keys = dict()
+    rel_keys = dict()
+    for element in graph:
+        if element.get("type") == "node":
+            if element.get("labels"):
+                if config.get("format").upper() == "TINKERPOP":
+                    node_keys.update({"labelV": get_type_string("labelV")})
+                else:
+                    node_keys.update(
+                        {"labels": get_type_string("labels")}
+                    )  # what if two nodes have same property name but different value type? or node's property name is "labels"?  # noqa: E501
```

Comment:
What does this mean? What will happen here?

Reply:
it will break things 🙃 because key id should be unique for every combination of key name and key type but since we are using key name as key id, it is not possible to have properties that have the same name and different value type because one will overwrite the other. I left it like that since neo4j did it like that also, I think they decided that it is a good tradeoff because it simplifies the process (by not using key ids) and rarely we have same property name for different type in same database?